### PR TITLE
ICU-22532 Set a value for _POSIX_C_SOURCE to get symlink() declared

### PR DESCRIPTION
**Pull request by roubert**
ISO C99 and later do not support implicit function declarations.

##### Checklist

- [X] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22532
- [X] Required: The PR title must be prefixed with a JIRA Issue number.
- [X] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [X] Required: Each commit message must be prefixed with a JIRA Issue number.
- [X] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: icu4c/source/tools/tzcode/zic.c [hunk 1]
```diff
@@ -3,6 +3,8 @@
 ** 2006-07-17 by Arthur David Olson.
 */
 
+#define _POSIX_C_SOURCE 200112L
```
Comment (FrankYFTang) [inline icu4c/source/tools/tzcode/zic.c:6]: could you add a comment about "get symlink() declared" ?

Comment (roubert) [reply icu4c/source/tools/tzcode/zic.c:6]: Done.

**Comment by roubert** [comment]
Do you have any further comments on this?

**Review by FrankYFTang — APPROVED** [review]
LGTM. thanks

**Pull request merged**